plugins/modules/systemd_info.py

```diff
@@ -0,0 +1,289 @@
+    then check if properties exist. If not, module fail.
+version_added: "10.4.0"
+options:
+  unitname:
+    description:
+      - List of unit names to process.
+    type: list
+    elements: str
+    default: []
+  extra_properties:
+    description:
+      - Additional properties to retrieve (appended to the default ones).
+    type: list
+    elements: str
+    default: []
+author:
+  - Marco Noce (@NomakCooper)
+'''
+
+EXAMPLES = r'''
+# Gather info for all systemd services, targets, sockets and mount
+- name: Gather all systemd unit info
+  community.general.systemd_info:
+  register: results
+
+# Gather info for selected units with extra properties.
+- name: Gather info for selected unit(s)
+  community.general.systemd_info:
+    unitname:
+      - systemd-journald.service
+      - systemd-journald.socket
+      - sshd-keygen.target
+      - -.mount
+    extra_properties:
+        - Description
+    register: results
+'''
+
+RETURN = r'''
+units:
+  description: Dictionary of systemd unit info keyed by unit name.
+  returned: success
+  type: dict
+  sample: {
+    "-.mount": {
+        "activestate": "active",
+        "description": "Root Mount",
+        "loadstate": "loaded",
+        "name": "-.mount",
+        "options": "rw,relatime,seclabel,attr2,inode64,logbufs=8,logbsize=32k,noquota",
+        "substate": "mounted",
+        "type": "xfs",
+        "what": "/dev/mapper/cs-root",
+        "where": "/"
+    },
+    "sshd-keygen.target": {
+        "activestate": "active",
+        "description": "sshd-keygen.target",
+        "fragmentpath": "/usr/lib/systemd/system/sshd-keygen.target",
+        "loadstate": "loaded",
+        "name": "sshd-keygen.target",
+        "substate": "active",
+        "unitfilepreset": "disabled",
+        "unitfilestate": "static"
+    },
+    "systemd-journald.service": {
+        "activestate": "active",
+        "description": "Journal Service",
+        "execmainpid": "613",
+        "fragmentpath": "/usr/lib/systemd/system/systemd-journald.service",
+        "loadstate": "loaded",
+        "mainpid": "613",
+        "name": "systemd-journald.service",
+        "substate": "running",
+        "unitfilepreset": "disabled",
+        "unitfilestate": "static"
+    },
+    "systemd-journald.socket": {
+        "activestate": "active",
+        "description": "Journal Socket",
+        "fragmentpath": "/usr/lib/systemd/system/systemd-journald.socket",
+        "loadstate": "loaded",
+        "name": "systemd-journald.socket",
+        "substate": "running",
+        "unitfilepreset": "disabled",
+        "unitfilestate": "static"
+    }
+  }
+'''
+
+from ansible.module_utils.basic import AnsibleModule
+
+
+def run_command(module, cmd):
+    rc, stdout, stderr = module.run_command(cmd, use_unsafe_shell=True)
+    if rc != 0:
+        module.fail_json(msg="Command '{0}' failed: {1}".format(" ".join(cmd), stderr.strip()))
```

Comment:
Two issues here, that I missed before:
* why `use_unsafe_shell=True`?
* instead of doing that rc validation you can simply:

```suggestion
    rc, stdout, stderr = module.run_command(cmd, check_rc=True)
```

Reply:
I used **`use_unsafe_shell=True`** as specified in the [Ansible documentation](https://docs.ansible.com/ansible/latest/dev_guide/developing_modules_best_practices.html).

[Module Security](https://docs.ansible.com/ansible/latest/dev_guide/developing_modules_best_practices.html#id11)
- **If you must use the shell, you must pass `use_unsafe_shell=True` to `module.run_command`.**

Reply:
> I used **`use_unsafe_shell=True`** as specified in the [Ansible documentation](https://docs.ansible.com/ansible/latest/dev_guide/developing_modules_best_practices.html).
> 
> [Module Security](https://docs.ansible.com/ansible/latest/dev_guide/developing_modules_best_practices.html#id11)
> - **If you must use the shell, you must pass `use_unsafe_shell=True` to `module.run_command`.**

The key thing being "if you must", which is not the case in your module. Nit sude how clear this is to you, the shell it mentions is /bin/bash (or a similar alternative), and it is only required if you need to use one of those shell features like `|`, or `>/somefile`, which is not the case. This module requires simply running a command, hence it should not set `use_unsafe_shell`.

Reply:
I misunderstood the documentation. I can **`remove use_unsafe_shell=True`** and leave only **`check_rc=True`**.